### PR TITLE
Make it possible to recompute the hazard maps and uhs from an existing calculation

### PR DESCRIPTION
**Pull request by micheles**
Closes https://github.com/gem/oq-engine/issues/2044. The demos are running here: https://ci.openquake.org/job/zdevel_oq-engine/2033/

To test:
1. run a demo producing hazard maps, for instance PointSourceClassicalPSHA
2. edit the job.ini and change the poes
3. re-run the demo, the new hazard maps will be regenerated without recomputing the probability curves

### CODE REVIEW

File: openquake/calculators/classical.py [hunk 1]
```diff
@@ -129,8 +161,7 @@ def __bool__(self):
         """
         True if the bounding box is non empty.
         """
-        return (self.min_dist is not None and self.west is not None and
-                self.south is not None)
+        return bool(self.min_dist and self.west and self.south)
```
Comment (ptormene) [inline openquake/calculators/classical.py]: Please add a comment about the fact we are not checking also north and east here.

File: openquake/calculators/classical.py [hunk 2]
```diff
@@ -274,32 +306,53 @@ def store_source_info(self, curves_by_trt_id):
...
+        Builds a dictionary curves_by_trt_gsim from the stored PoEs
+        """
+        curves_by_trt_gsim = {}
+        with self.monitor('read poes', autoflush=True):
+            for group_id in self.datastore['poes']:
+                grp_id = int(group_id)
+                poes = self.datastore['poes/' + group_id]
+                gsims = self.rlzs_assoc.gsims_by_grp_id[grp_id]
+                for i, gsim in enumerate(gsims):
+                    curves_by_trt_gsim[grp_id, gsim] = poes.extract(i)
```
Comment (ptormene) [inline openquake/calculators/classical.py]: Please check the name

File: openquake/calculators/classical.py [hunk 3]
```diff
@@ -274,32 +306,53 @@ def store_source_info(self, curves_by_trt_id):
...
+        return curves_by_trt_gsim
+
+    def post_execute(self, curves_by_trt_gsim):
```
Comment (ptormene) [inline openquake/calculators/classical.py]: Same as above

File: openquake/calculators/tests/classical_test.py [hunk 4]
```diff
@@ -138,6 +138,12 @@ def test_case_13(self):
              'hazard_map-mean.csv'],
             case_13.__file__)
 
+        # test recomputing the hazard maps
```
Comment (ptormene) [inline openquake/calculators/tests/classical_test.py]: Please check if the hazard calculation id is missing

**Comment by ptormene** [comment]
Besides some comments, LGTM

File: openquake/calculators/classical.py (hunk 1 above)
Comment (micheles) [inline openquake/calculators/classical.py]: There was an error in the logic that I fixed, good catch!

File: openquake/calculators/classical.py (hunk 2 above)
Comment (micheles) [inline openquake/calculators/classical.py]: Renamed curves_by_trt_gsim -> curves_by_grp_gsim

File: openquake/calculators/tests/classical_test.py (hunk 4 above)
Comment (micheles) [inline openquake/calculators/tests/classical_test.py]: It was missing indeed. Fixed.

**Pull request merged**